misc/mdns/src/service.rs

```diff
@@ -147,33 +153,60 @@ impl MdnsService {
         Ok(MdnsService {
             socket,
             query_socket: UdpSocket::bind((Ipv4Addr::from([0u8, 0, 0, 0]), 0u16)).await?,
-            query_interval: Interval::new(Duration::from_secs(20)),
+            query_interval: Interval::new_at(Instant::now(), Duration::from_secs(20)),
             silent,
             recv_buffer: [0; 2048],
             send_buffers: Vec::new(),
             query_send_buffers: Vec::new(),
         })
     }
 
-    pub async fn next_packet(&mut self) -> MdnsPacket {
-        // TODO: refactor this block
+    pub fn enqueue_response(&mut self, rsp: Vec<u8>) {
```

Comment:
I don't think this is supposed to be public, right?

```suggestion
    fn enqueue_response(&mut self, rsp: Vec<u8>) {
```

Reply:
This function is called within `Mdns` `Networkbehaviour` implementation, thus it needs to be `pub`.